src/witan/send/chart.clj

```diff
@@ -317,9 +317,12 @@
         series))
 
 (defn comparison-chart-and-table [comparison-defs]
-  (-> comparison-defs
-      (assoc :chart (comparison-chart comparison-defs))
-      (assoc :table (comparison-table comparison-defs))))
+  (try
+    (-> comparison-defs
+        (assoc :chart (comparison-chart comparison-defs))
+        (assoc :table (comparison-table comparison-defs)))
+    (catch Exception e
+      (throw (ex-info "Unable to create comparison table and charts" {:comparison-defs comparison-defs} e)))))
```

Comment:
I suppose throwing `comparison-defs` will show you where there's something missing/nil?

Reply:
yeah, I was able to see that the history and projection vectors were empty so I knew that the data was getting lost somewhere to this point.

Reply:
👍